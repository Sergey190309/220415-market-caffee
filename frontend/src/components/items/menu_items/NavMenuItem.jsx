import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

// import styled from 'styled-components'
import {
  MenuItem
  // Icon
  // Grid, Button
} from 'semantic-ui-react'

// const StyledNavMenuItem = styled(MenuItem)`
//   font-weight: bold;
//   font-size: 1.2em;
// `

const NavMenuItem = ({ disabled, name, to, active, content, onClick }) => {
  // console.log('NavMenuItem:\n',
  //   '\n  disabled ->', disabled
  // )
  return (
    <MenuItem
      style={{ fontWeight: 'bold', fontSize: '1.2em' }}
      // fitted='horizontally'
      as={disabled ? null : Link}
      disabled={disabled}
      name={name}
      to={to}
      active={active}
      content={content}
      onClick={onClick}
    />
    //   {content}
    //   {/* {children} */}
    // </MenuItem>
  )
}

NavMenuItem.defaultProps = {
  // link: {},
  disabled: false,
  name: '',
  to: '',
  active: false,
  onClick: () => { },
  content: ''
}

NavMenuItem.propTypes = {
  // children: PropTypes.object.isRequired,
  // link: PropTypes.object.isRequired,
  disabled: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  content: PropTypes.string.isRequired
}
export default NavMenuItem
