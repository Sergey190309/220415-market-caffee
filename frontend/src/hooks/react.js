import {useState, useEffect, useRef, useCallback } from 'react'

export const useAppState = useState
export const useAppEffect = useEffect
export const useAppRef = useRef
export const useAppCallback = useCallback
