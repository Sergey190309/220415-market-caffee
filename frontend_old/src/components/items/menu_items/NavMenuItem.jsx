import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

// import styled from 'styled-components'
import {
  MenuItem
  // Icon
  // Grid, Button
} from 'semantic-ui-react'

// const StyledNavMenuItem = styled(MenuItem)`
//   font-weight: bold;
//   font-size: 1.2em;
// `

const NavMenuItem = ({
  disabled,
  name,
  to,
  active,
  content,
  onClick
}) => {
  // console.log('NavMenuItem:\n',
  //   '\n  disabled ->', disabled
  // )
  return (
    <MenuItem
      data-testid='MenuItem'
      style={{ fontWeight: 'bold', fontSize: '1.2em' }}
      as={disabled ? null : Link}
      disabled={disabled}
      name={name}
      to={to}
      active={active}
      content={content}
      onClick={onClick}
    />
  )
}

NavMenuItem.defaultProps = {
  disabled: false,
  name: '',
  to: '',
  active: false,
  onClick: () => { },
  content: ''
}

NavMenuItem.propTypes = {
  disabled: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  content: PropTypes.string.isRequired
}
export default NavMenuItem
