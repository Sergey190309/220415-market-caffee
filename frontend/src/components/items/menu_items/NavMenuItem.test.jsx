import React from 'react'
import { screen } from '@testing-library/react'

import NavMenuItem from './NavMenuItem'
import { testLinkedRender } from '../../../tstHelpers'
import userEvent from '@testing-library/user-event'
describe('NavMenuItem', () => {
  const testProps = {
    disabled: false,
    name: 'mockName',
    to: 'mockUrl',
    active: true,
    content: 'mockContent',
    onClick: jest.fn()
  }
  describe('appearance', () => {
    test('normal rendering (snapshot)', () => {
      const activeProps = { ...testProps }
      testLinkedRender(<NavMenuItem {...activeProps} />)
      const menuItem = screen.getByTestId('MenuItem')
      expect(menuItem).toMatchSnapshot()
      expect(menuItem).toHaveClass('active')
      expect(menuItem).toHaveAttribute('href', `/${activeProps.to}`)
      expect(menuItem).toHaveTextContent(activeProps.content)
      // screen.debug(menuItem)
    })
    test('rendering, disabled, not active (snapshot)', () => {
      const activeProps = { ...testProps, active: false, disabled: true }
      testLinkedRender(<NavMenuItem {...activeProps} />)
      const menuItem = screen.getByTestId('MenuItem')
      expect(menuItem).not.toHaveClass('active')
      expect(menuItem).not.toHaveAttribute('href')
      expect(menuItem).toHaveAttribute('to', activeProps.to)
      // screen.debug(menuItem)
    })
  })
  describe('actions', () => {
    test.skip('click', () => {
      /**
       * Unable to find the "window" object for the given node.
       * Producer offer to report to there project.
       */
      const activeProps = { ...testProps }
      testLinkedRender(<NavMenuItem {...activeProps} />)
      const menuItem = screen.getByTestId('MenuItem')
      userEvent.click('menuItem')
      // expect(activeProps.onClick).toHaveBeenCalledTimes(1)
      // expect(menuItem).toHaveClass('active')
      // expect(menuItem).toHaveAttribute('href', `/${activeProps.to}`)
      // expect(menuItem).toHaveTextContent(activeProps.content)
      screen.debug(menuItem)
    })
  })
})
