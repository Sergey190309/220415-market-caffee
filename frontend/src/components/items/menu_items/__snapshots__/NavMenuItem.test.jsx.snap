// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenuItem appearance normal rendering (snapshot) 1`] = `
<a
  class="active item"
  data-testid="MenuItem"
  href="/mockUrl"
  style="font-weight: bold; font-size: 1.2em;"
>
  mockContent
</a>
`;
